drivers/sensor/vl53l1x/vl53l1_platform_user_data.h

```diff
@@ -0,0 +1,83 @@
+/* vl53l1_platform_user_data.h - Zephyr customization of ST vl53l1x library. */
+
+/*
+ * Copyright (c) 2017 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_VL53L1X_VL53L1_PLATFORM_USER_DATA_H_
+#define ZEPHYR_DRIVERS_SENSOR_VL53L1X_VL53L1_PLATFORM_USER_DATA_H_
+
+#include "vl53l1_def.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct VL53L1_Dev_t
+ * @brief  Generic PAL device type that does link between API and platform
+ * abstraction layer
+ *
+ */
+typedef struct {
+	/*!< Low Level Driver data structure */
+	VL53L1_DevData_t Data;
+	/*!< New data ready poll duration in ms - for debug */
+	uint32_t new_data_ready_poll_duration_ms;
+	/*!< I2C device handle */
+	const struct i2c_dt_spec *i2c;
+} VL53L1_Dev_t;
+
+
+/**
+ * @brief   Declare the device Handle as a pointer of the structure @a VL53L1_Dev_t.
+ *
+ */
+typedef VL53L1_Dev_t *VL53L1_DEV;
```

Comment:
Same here, please just use a struct pointer

Reply:
@cfriedt These typedefs are used throughout the STM library being used by this driver thus cannot be changed. There is a similar driver for a different time-of-flight sensor (see drivers/sensor/vl53l0x) that uses a similar library by STM and also has these typedefs, but doesn't appear to be an issue since it is in the main zephyr tree. I spoke to @MaureenHelm about this and she agreed to override the failing compliance checks.

Reply:
Ah, in that sense, ok